src/metabase/api/dashboard.clj

```diff
@@ -604,14 +606,17 @@
        ;; TODO - we should combine these all into a single UNION ALL query against the data warehouse instead of doing a
        ;; separate query for each Field (for parameters that are mapped to more than one Field)
       (try
-        (let [results (distinct (mapcat (if (seq query)
-                                          #(chain-filter/chain-filter-search % constraints query :limit result-limit)
-                                          #(chain-filter/chain-filter % constraints :limit result-limit))
-                                        field-ids))]
+        (let [results (map (if (seq query)
```

Comment:
any Clojure magic trick to do this without building an immediate concatnated results?